californium-core/src/main/java/org/eclipse/californium/core/network/EndpointManager.java

```diff
@@ -358,9 +258,12 @@ public void deliverRequest(Exchange exchange) {
 
 		@Override
 		public void deliverResponse(Exchange exchange, Response response) {
-			if (exchange == null) throw new NullPointerException();
-			if (exchange.getRequest() == null) throw new NullPointerException();
-			if (response == null) throw new NullPointerException();
+			if (exchange == null)
+				throw new NullPointerException("no CoAP exchange!");
```

Comment:
True, the question is more: do we expect that this method will ever be invoked with null argument(s), given that it is basically an internal API only? I have no clue, though, whether this can actually happen ;-)